Content.Shared/Paint/PaintRemoverSystem.cs

```diff
@@ -0,0 +1,67 @@
+using Content.Shared.Popups;
+using Content.Shared.Interaction;
+using Content.Shared.DoAfter;
+using Robust.Shared.Audio.Systems;
+using Robust.Shared.Timing;
+
+namespace Content.Shared.Paint;
+
+/// <summary>
+/// Removes paint from an entity.
+/// </summary>
+public sealed class PaintRemoverSystem : SharedPaintSystem
+{
+    [Dependency] private readonly SharedPopupSystem _popup = default!;
+    [Dependency] private readonly SharedDoAfterSystem _doAfter = default!;
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+    [Dependency] private readonly IGameTiming _timing = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<PaintRemoverComponent, AfterInteractEvent>(OnInteract);
+        SubscribeLocalEvent<PaintRemoverComponent, PaintRemoverDoAfterEvent>(OnDoAfter);
+    }
+
+    // When entity is painted, remove paint from that entity.
+    private void OnInteract(EntityUid uid, PaintRemoverComponent component, AfterInteractEvent args)
+    {
+        if (args.Handled)
+            return;
+
+        if (!args.CanReach || args.Target is not { Valid: true } target || !HasComp<PaintedComponent>(target))
+            return;
+
+        _doAfter.TryStartDoAfter(new DoAfterArgs(EntityManager, args.User, component.CleanDelay, new PaintRemoverDoAfterEvent(), uid, args.Target, uid)
+        {
+            BreakOnUserMove = true,
+            BreakOnDamage = true,
+            BreakOnTargetMove = true,
+            MovementThreshold = 1.0f,
+        });
+        args.Handled = true;
+    }
+
+    private void OnDoAfter(EntityUid uid, PaintRemoverComponent component, DoAfterEvent args)
+    {
+        if (args.Cancelled || args.Handled || args.Args.Target == null)
+            return;
+
+        if (args.Target is not { Valid: true } target)
+            return;
+
+        if (!TryComp(target, out PaintedComponent? paint))
+            return;
+
+        if (_timing.IsFirstTimePredicted)
+        {
+            paint.Enabled = false;
+            _audio.PlayPvs(component.Sound, target);
+            _popup.PopupClient(Loc.GetString("you clean off the paint", ("target", target)), args.User, args.User, PopupType.Medium);
+            RemComp<PaintedComponent>(target);
+            Dirty(target, paint);
+        }
+        args.Handled = true;
```

Comment:
IsFirstTimePredicted is unneeded here, just replace playpvs with playpredicted.

Reply:
this is done